src/pages/Signup.jsx

```diff
@@ -1,177 +1,181 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
+import { useStateCallback } from "use-state-callback";
 import { useNavigate } from "react-router-dom";
-import { registerUser } from "../state/util/index";
+import { useUser, validateSignUp } from "../state/index.js";
+import { toast } from "react-toastify";
+import { registerUser } from "../state/util/registerUser.js";
 
 const Signup = () => {
   const navigate = useNavigate();
-  const [signUpForm, updateSignUpForm] = useState({
-    email: "   ",
-    password: "",
-    confirmPassword: "",
-    firstName: "",
-    lastName: "",
-  });
 
-  const [validationMessage, setValidationMessage] = useState({
-    email: "",
-    password: "",
-    confirmPassword: "",
-    firstName: "",
-    lastName: "",
-  });
+  const [firstName, setFirstName] = useStateCallback("");
+  const [lastName, setLastName] = useStateCallback("");
+  const [email, setEmail] = useStateCallback("");
+  const [password, setPassword] = useStateCallback("");
+  const [confirmPassword, setConfirmPassword] = useStateCallback("");
+  //error message
+  const [firstNameError, setFirstNameError] = useState("");
+  const [lastNameError, setLastNameError] = useState("");
+  const [emailError, setEmailError] = useState("");
+  const [passwordError, setPasswordError] = useState("");
+  const [confirmPasswordError, setConfirmPasswordError] = useState("");
+  const [showPassword, setShowPassword] = useState(false);
+  const [showConfirmPassword, setShowConfirmPassword] = useState(false);
```

Comment:
As you are using so many states you can use reducer instead of usestate

Reply:
same suggestion , u can use reducer